2/contrib/s2i/run

```diff
@@ -105,6 +105,28 @@ function migrate_slave_configuration() {
   sed -i "s#<image>registry.access.redhat.com/openshift3/jenkins-slave-nodejs-rhel7\(:.*\)\{0,1\}</image>#<image>registry.access.redhat.com/openshift3/jenkins-slave-nodejs-rhel7:${JENKINS_SLAVE_IMAGE_TAG}</image>#" /var/lib/jenkins/config.xml
 }
 
+function force_copy_plugins() {
+   # if rhel
+   if [ "$(ls /usr/lib/jenkins/*.hpi 2>/dev/null)" ]; then
+      echo "This openshift jenkins image is starting in a deployment where the image version"
+      echo "was not recoreded.  A one time replacement of all the plugins from the image to the"
+      echo "jenkins plugin directory will occur."
+      echo ""
+      ls -la /usr/lib/jenkins
+      for FILENAME in /usr/lib/jenkins/*.hpi ; do
+         basefilename=`basename $FILENAME .hpi`
+         plugin_image_volume_location="${JENKINS_HOME}/plugins/${basefilename}.jpi"
+         echo "Force linking ${FILENAME} to ${plugin_image_volume_location}"
+         ln -sf $FILENAME $plugin_image_volume_location
```

Comment:
I guess having symlinks becomes somewhat moot in a world where you're always overwriting existing plugins w/ the plugins from the image, if the image is new (and hopefully not overwriting anything if the image is not new).  So does it make sense to make this a straight copy, so that this and the centos image behavior are aligned?

Reply:
Yeah I believe your comment below answers this one.  But certainly talking through it face to face to be sure may be prudent here.